### ISSUE TITLE
Fedora packages or COPR repo

### ISSUE THREAD

**Issue by rapgro** [OpenRA/OpenRA#8264]
Fedora has strange issues to provide an usable Mono environment, at least for Mono versions 3 and 4.
Request is to provide all OpenRA packages incl. unbundling at COPR via a clean repository. SUSE OpenBuild Service (OBS) can not deal with partly COPR based packages. So let's move the whole stuff, at least for Fedora concerns, to COPR.

See also:
https://github.com/OpenRA/OpenRA/issues/7140
https://github.com/OpenRA/OpenRA/issues/6863
##

**Comment by rapgro** [comment]
There are semi-official builds at Copr for both Mono 3 and 4. Maybe use those repos to provide official Fedora assemblies from OBS?
Mono4 https://copr.fedoraproject.org/coprs/tpokorra/mono/
Mono3 https://copr.fedoraproject.org/coprs/elsupergomez/mono/

**Comment by Mailaender** [comment]
:+1: 

I currently maintain it at https://build.opensuse.org/project/show/games:openra but as I don't run Fedora myself those packages are completely untested. Started to unbundle some dependencies, but I haven't finished the quest yet. I also know few about the Fedora packaging specialties so I am glad an expert can take over and relieve me of it. Once you are done, we may want to advertise your repositories at https://github.com/OpenRA/OpenRAWeb.

You will run into compilation errors from time to time as we don't test our builds against Mono 2.10 anymore and got a WONTFIX from @Xamarin on https://bugzilla.xamarin.com/show_bug.cgi?id=30256 However the patches are usually simple and we will try to fix those compilation errors due to oversights in our code ASAP and can help you there.

I was told that Mono 4 is currently work in progress and @tpokorra is currently updating everything at https://github.com/tpokorra/lbs-mono-fedora It was even noteworthy https://www.phoronix.com/scan.php?page=news_item&px=Mono-4-Fedora-23

**Comment by rapgro** [comment]
Thank for your support. IMHO it's not realistic to expect Mono 4 in a good state and for Fedora 23, because of all the needed bootstrapping from discontinued Mono 2 binaries. Maybe it's a good decision to build OpenQA packages for Mono 3 at the moment.
https://fedoraproject.org/wiki/Changes/Mono_4

**Comment by rapgro** [comment]
https://github.com/OpenRA/OpenRA/issues/9254

**Comment by rapgro** [comment]
https://fedoraproject.org/wiki/Changes/Mono_4
This says we can expect Mono 4 in Fedora 23. There's no need to continue with Copr.

**Comment by rapgro** [comment]
Package requested: https://bugzilla.redhat.com/show_bug.cgi?id=1159091

I'm still very busy with those review requests and could need some more helpy hands.
Gentlemen please?

**Comment by Mailaender** [comment]
I started building CentOS packages at https://build.opensuse.org/project/show/games:openra however it is very annoying. Even Fedora and CentOS are incompatible. I am getting ridiculous fails due to `%changelog` date formatting etc. not to mention all the missing Mono requirements and package renamings and path changes so I might not finish this as I don't personally use Fedora at all nor do know anyone who does.

**Comment by Mailaender** [comment]
You are actually doing much better than I am at building the dependencies. The build systems of our dependencies are often non-existing or completely rotten for Mono. I have several question on how you solved things. Let's meet in IRC at some point to exchange packaging fixes and best practices.

**Comment by Aricg** [comment]
Got it to compile on fedora23. Needed the stable so that I could play online.
Steps:
installed openra from @Mailaender 's suse repo, could only grab the playtest this way. 
install nuget-devel and wine-openal
git show-ref
git checkout 0ac7b73d37edd18d13efb9e8a9d5ecc9be7e711f
make clean && make dependencies && make version && make all

I can help package this for copr.

**Comment by rapgro** [comment]
@Aricg Hi, thanks for your interest. Copr is obsolete and currently doesn't bring any benefit over OBS. We're working on official packages (with all mono version 4 dependencies) in Fedora repository. It would be great if you can help there instead.
Again: Please look into the linked bugs ("Depends"), those are to unbundle all 3rdparty libraries, most of them are still as a placeholder only:
https://bugzilla.redhat.com/show_bug.cgi?id=1159091

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/issues/10280 will reduce RestSharp from that list and give us a system installable MaxMind.GeoIP.

**Comment by Mailaender** [comment]
Another attempt to switch to Open.NAT https://github.com/OpenRA/OpenRA/pull/10281.

**Comment by rapgro** [comment]
Hmm, because of the underrated amount of needed bureaucracy (which I do not critizise in general due to it has its sensefulness of course!) in Fedora, I currently think again a Copr makes more sense ATM. Also we miss people interested in Fedora mono and to do all the reviews for official packages incl. 3rd party unbundling. Myself has also very few spare time only to work on this process.

**Comment by rapgro** [comment]
Is it possible to replace fuzzynet with DotFuzzy [1]? fuzzynet has an unclear license, there's no possibility to work around this for a Fedora package [2]. DotFuzzy has LGPLv3 and therefore has a bigger chance to get packaged properly. The ruleset language is very similiar except uppercase spelling for the keywords.
[1] https://github.com/MicheleBertoli/DotFuzzy
[2] https://bugzilla.redhat.com/show_bug.cgi?id=1166916

**Comment by pchote** [comment]
I remember that in one of the earlier topics about this you/we established that fuzzynet is definitely GPL.  How about you fork it, add the file headers, and then package that fork to pass fedora's lint checks?

**Comment by rapgro** [comment]
> Packagers should point out to upstream that by not including a proper full license text, they are making it difficult or impossible for anyone to comply with their desired license terms. 
https://fedoraproject.org/wiki/Packaging:LicensingGuidelines#License_Text

**Comment by pchote** [comment]
> Anyways, GPL forbids to bundle it as a library in general.

Can you clarify this?

**Comment by rapgro** [comment]
https://www.gnu.org/licenses/why-not-lgpl.en.html

**Comment by Mailaender** [comment]
This has been closed upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1159091 and the current proposal https://github.com/OpenRA/OpenRA/issues/12554 is also going in an opposite direction so unbundling of thirdparty libraries and a clear separation of an engine built on freely licensed source code and not so free content with unclear redistribution rights isn't really in sight. It is a bit sad that I would not only like to see this in Fedora but also on https://software.opensuse.org/packages/Games to close the circle as I found out about OpenRA as a Novell employee submitted a package to build.opensuse.org back in the day, but I have given up hope on that now as well.

**Comment by Mailaender** [comment]
I propose https://github.com/OpenRA/OpenRA/issues/12663 as an alternative.

**Comment by pchote** [comment]
I'm rather angry to see @rapgro drop this with such predjudice after the work we have put in over the years to try and support Fedora.  We wasted a lot of time rewriting our dependency fetching and compilation setup (more than once) for the sake of their arbitrary bundling rules, and we took the flack for dropping Windows XP support largely because we wanted to replace Mono.NAT with Open.NAT and unblock the Fedora packaging again.

It looks like the final straw was that I was "unwilling to accept proposed solutions" by questioning the latest demand to replace one of our long-stable dependencies with an untested and unmaintained replacement.  I'm not sure about the denying patches part, because I can't find any examples of rejecting code from him.

**Comment by lupinix** [comment]
fuzzynet: https://github.com/kaluzhny/fuzzynet by original dev, with GPL license file.

**Comment by rapgro** [comment]
Fuzzynet ist without a license text. There's no possibility to include such projects as official packages in Fedora. How to explain better what GPL is than to give license text? I talked with FSFE about that issue, they say I'm right in the decision to stop the project. Please don't expect any further comment from my side on this issue, my free time is too costly. Anyone else interested, feel free to continue my started work.

**Comment by Mailaender** [comment]
https://sourceforge.net/projects/fuzzynet/ also states GPL 2.0 so this was just an oversight by the original developer when creating the source code ZIP file.
